src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionListener.java

```diff
@@ -0,0 +1,45 @@
+package org.jenkinsci.plugins.workflow.flow;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+
+/**
+ * Listens for significant status updates for a {@link FlowExecution}, such as started running, paused, or completed.
+ *
+ * @since 2.14
+ * @author Andrew Bayer
+ */
+public abstract class FlowExecutionListener implements ExtensionPoint {
+
+    /**
+     * Called when a {@link FlowExecution} has started running or resumed.
+     *
+     * @param execution The {@link FlowExecution} that has started running or resumed.
```

Comment:
Ok? Still not sure how to distinguish between start/resume calls in `workflow-cps`, but I'll add a boolean here, I guess.